builders/tutorials/subsquid.md

```diff
@@ -0,0 +1,490 @@
+
+```bash
+sqd codegen
+```
+
+In the next step, we'll use the ERC-20 ABI to automatically generate TypeScript interface classes. Below is a generic ERC-20 standard ABI. Copy and paste it into a file named `erc20.json` in the abi folder at the root level of the project. 
+
+??? code "View the complete ABI"
+
+    ```json
+    --8<-- 'code/tutorials/erc20.json'
+    ```
+
+
+
+Next, we can use our contract's ABI to generate TypeScript interface classes. We can do this by running:
+
+```bash
+sqd typegen
+```
+
+![Run Subsquid commands](/images/builders/tutorials/subsquid/subsquid-3.png)
+
+This will generate the related TypeScript interface classes in the `src/abi/erc20.ts` file. For this tutorial, we'll be accessing the `events` specifically.
+
+### Configure the Processor {: #configure-the-processor}  
+
+The `processor.ts` file tells Subsquid exactly what data you'd like to ingest. Transforming that data into the exact desired format will take place at a later step. In `processor.ts`, we'll need to indicate a data source, contract address, the event(s) to index, and a block range. 
+
+Open up the `src` folder and head to the `processor.ts` file. You'll see the line `export const processor = new EvmBatchProcessor()` followed by `.setDataSource`. We'll need to make a few changes here. Subsquid has [available archives for many chains](https://docs.subsquid.io/evm-indexing/supported-networks/){target=_blank} that can speed up the data retrieval process, but it's unlikely your containerchain has a hosted archive already. But not to worry, Subsquid can easily get the data it needs via your ContainerChain's RPC URL. Go ahead and comment out or delete the archive line. Once done, your code should look similar to the below:
+
+```ts
+    .setDataSource({
+        // Lookup archive by the network name in Subsquid registry
+        // See https://docs.subsquid.io/evm-indexing/supported-networks/
+        chain: {
+            url: assertNotNull(process.env.RPC_ENDPOINT),
+            rateLimit: 300
+        }
+    })
+```
+
+You'll need to provide the RPC URL for your ContainerChain in your `.env` file. Make sure that any existing RPC URLs are removed. If you're using the Demo EVM ContainerChain, the respective line in your .env file will look like this: 
+
+```
+RPC_ENDPOINT={{ networks.dancebox.rpc_url }}
+```
+
+Of course, we need to tell the Subsquid processor which contract we're interested in. Create a constant for the address in the following manner: 
+
+```
+export const CONTRACT_ADDRESS = 'INSERT_CONTRACT_ADDRESS'.toLowerCase();
+```
+
+The `.toLowerCase()` is critical because the Subsquid processor is case sensitive, and some block explorers format contract addresses with capitalization. Now, let's define the event that we want to index by adding the following: 
+
+```ts
+    .addLog({
+        address: [CONTRACT_ADDRESS],
+        topic0: [erc20.events.Transfer.topic],
+        transaction: true,
```

Comment:
what does transaction true do?

Reply:
I think it signifies that you want to include information about the corresponding txn like hash, gas, sender, etc. Good for debugging purposes even if you don't need that data in your model. 
If set to false then it will skip including that info.